### ISSUE TITLE
find_gene_modules() in monocle3 return "Error in array  : 'data' must be of a vector type, was 'NULL'"

### ISSUE THREAD

**Issue by godzillaStudios** [cole-trapnell-lab/monocle-release#439]
Hello, 

I tried to call find_gene_modules(), but it returns "Error in array(x, c(length(x), 1L), if (!is.null(names(x))) list(names(x),  : 'data' must be of a vector type, was 'NULL'". 
Traceback:

6.array(x, c(length(x), 1L), if (!is.null(names(x))) list(names(x), NULL) else NULL)
5.as.matrix.default(preprocess_mat)
4.as.matrix(preprocess_mat)
3. | uwot(X = X, n_neighbors = n_neighbors, n_components = n_components, metric = metric, n_epochs = n_epochs, alpha = learning_rate, scale = scale, init = init, init_sdev = init_sdev, spread = spread, min_dist = min_dist, set_op_mix_ratio = set_op_mix_ratio, ...
2. | uwot::umap(as.matrix(preprocess_mat), n_components = max_components, metric = umap.metric, min_dist = umap.min_dist, n_neighbors = umap.n_neighbors, fast_sgd = umap.fast_sgd, n_threads = cores, verbose = verbose, nn_method = umap.nn_method, ...)
1.find_gene_modules(cds_subset[pr_deg_ids, ], resolution = 0.001)

I didn't find any non-numeric data (like NA or NaN, etc.) in the data. And I cannot find any solution for this:(. Could you help me out with this? Please. 

 I'm looking forward to hearing from you. Thank you.

My code is as follows.
`
cds_subset <- choose_cells(cds) 

subset_pr_test_res <- graph_test(cds_subset, neighbor_graph="principal_graph",reduction_method = "UMAP" ,cores=1)
pr_deg_ids <- row.names(subset(subset_pr_test_res, q_value < 0.05))
gene_module_df <- find_gene_modules(cds_subset[pr_deg_ids,], resolution=0.001)`

You can find my cds_subset file from this [link](https://drive.google.com/file/d/16EwVp6ASrFJ9-8vTKG__qms9WkznsfDH/view?usp=sharing)

My session info as follows.

> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
LAPACK: /Users/mike/opt/anaconda3/envs/R/lib/libopenblasp-r0.3.12.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] shiny_1.5.0                 magrittr_2.0.1              patchwork_1.1.1            
 [4] ggplot2_3.3.3               SeuratWrappers_0.3.0        SeuratData_0.2.1           
 [7] Seurat_3.2.3                monocle3_0.2.1              SingleCellExperiment_1.8.0 
[10] SummarizedExperiment_1.16.0 DelayedArray_0.12.0         BiocParallel_1.20.1        
[13] matrixStats_0.57.0          GenomicRanges_1.38.0        GenomeInfoDb_1.22.0        
[16] IRanges_2.20.2              S4Vectors_0.24.4            Biobase_2.46.0             
[19] BiocGenerics_0.32.0        

loaded via a namespace (and not attached):
  [1] plyr_1.8.6             igraph_1.2.6           lazyeval_0.2.2         sp_1.4-5              
  [5] splines_3.6.3          listenv_0.8.0          scattermore_0.7        digest_0.6.27         
  [9] htmltools_0.5.1.1      viridis_0.5.1          rsconnect_0.8.16       gdata_2.18.0          
 [13] fansi_0.4.2            tensor_1.5             cluster_2.1.0          ROCR_1.0-11           
 [17] remotes_2.2.0          globals_0.14.0         gmodels_2.18.1         colorspace_2.0-0      
 [21] rappdirs_0.3.1         ggrepel_0.9.1          xfun_0.20              dplyr_1.0.3           
 [25] crayon_1.3.4           RCurl_1.98-1.2         jsonlite_1.7.2         spatstat_1.64-1       
 [29] spatstat.data_1.7-0    survival_3.2-7         zoo_1.8-8              glue_1.4.2            
 [33] polyclip_1.10-0        gtable_0.3.0           zlibbioc_1.32.0        XVector_0.26.0        
 [37] leiden_0.3.6           future.apply_1.7.0     abind_1.4-5            scales_1.1.1          
 [41] DBI_1.1.1              miniUI_0.1.1.1         Rcpp_1.0.6             viridisLite_0.3.0     
 [45] xtable_1.8-4           spData_0.3.8           units_0.6-7            reticulate_1.18       
 [49] spdep_1.1-5            rsvd_1.0.3             proxy_0.4-24           htmlwidgets_1.5.3     
 [53] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.1         ica_1.0-2             
 [57] pkgconfig_2.0.3        farver_2.0.3           uwot_0.1.10            deldir_0.2-9          
 [61] tidyselect_1.1.0       labeling_0.4.2         rlang_0.4.10           reshape2_1.4.4        
 [65] later_1.1.0.1          pbmcapply_1.5.0        munsell_0.5.0          tools_3.6.3           
 [69] cli_2.2.0              generics_0.1.0         ggridges_0.5.3         stringr_1.4.0         
 [73] fastmap_1.0.1          yaml_2.2.1             goftest_1.2-2          knitr_1.30            
 [77] fitdistrplus_1.1-3     purrr_0.3.4            RANN_2.6.1             pbapply_1.4-3         
 [81] future_1.21.0          nlme_3.1-151           mime_0.9               slam_0.1-48           
 [85] compiler_3.6.3         rstudioapi_0.13        plotly_4.9.3           png_0.1-7             
 [89] e1071_1.7-4            spatstat.utils_2.0-0   tibble_3.0.5           stringi_1.5.3         
 [93] lattice_0.20-41        Matrix_1.3-2           classInt_0.4-3         vctrs_0.3.6           
 [97] LearnBayes_2.15.1      pillar_1.4.7           lifecycle_0.2.0        BiocManager_1.30.10   
[101] lmtest_0.9-38          RcppAnnoy_0.0.18       data.table_1.13.6      cowplot_1.1.1         
[105] bitops_1.0-6           irlba_2.3.3            raster_3.4-5           httpuv_1.5.5          
[109] R6_2.5.0               promises_1.1.1         KernSmooth_2.23-18     gridExtra_2.3         
[113] parallelly_1.23.0      codetools_0.2-18       gtools_3.8.2           boot_1.3-25           
[117] MASS_7.3-53            assertthat_0.2.1       leidenbase_0.1.2       withr_2.4.0           
[121] sctransform_0.3.2      GenomeInfoDbData_1.2.2 expm_0.999-5           mgcv_1.8-33           
[125] grid_3.6.3             rpart_4.1-15           coda_0.19-4            class_7.3-18          
[129] tidyr_1.1.2            Rtsne_0.15             sf_0.9-7

**Comment by godzillaStudios** [comment]
I imported Seurat object into monocle3 (cds). This issue can be fixed by using the Seurat object before renaming the cluster.
